Clarify the release steps (#210)
meilisearch/meilisearch-java: CONTRIBUTING.md

⚠️ Before doing anything, make sure you got through the guide about [Releasing an Integration](https://github.com/meilisearch/integration-guides/blob/main/guides/integration-release.md).

Make a PR modifying the following files with the right version:

- [`build.gradle`](/build.gradle)

```java
version = 'X.X.X'
```

- [`README.md`](/README.md) in the `Installation` section

```xml
<version>X.X.X</version>
```

```groovy
implementation 'com.meilisearch.sdk:meilisearch-java:X.X.X'
```

Once the changes are merged on `main`, you can publish the current draft release via the [GitHub interface](https://github.com/meilisearch/meilisearch-java/releases).

A GitHub Action will be triggered and publish a new release to Maven repository.

### How to Manually Publish the Release to Maven repository  <!-- omit in TOC -->

⚠️ These following steps should only be applied if it's impossible to release the current version in the Maven repository via the CI. Keep in mind publishing via the CI should always be privileged.

#### Create signature credentials (first time) <!-- omit in TOC -->

> ⚠️ All these steps (create and publish a GPG key) have already been done by the Meili team and the key is shared internally. Please ask a maintainer to get the credentials if needed.

Reviewer: ```suggestion
⚠️ These step (create and publish a GPG key) has already been done by the Meili team and the key is shared internally. Please ask a maintainer to get the credentials if needed.
```

⚠️ Before doing anything, make sure you got through the guide about [Releasing an Integration](https://github.com/meilisearch/integration-guides/blob/main/guides/integration-release.md).

> #### Update the version <!-- omit in TOC -->
> 

Reviewer: ```suggestion
```

Make a PR modifying the following files with the right version:

- [`build.gradle`](/build.gradle)

```java
version = 'X.X.X'
```

- [`README.md`](/README.md) in the `Installation` section

```xml
<version>X.X.X</version>
```

```groovy
implementation 'com.meilisearch.sdk:meilisearch-java:X.X.X'
```

Once the changes are merged on `main`, you can publish the current draft release via the [GitHub interface](https://github.com/meilisearch/meilisearch-java/releases).

A GitHub Action will be triggered and publish a new release to Maven repository.

> ### How to Publish the Release to Maven repository manually if the CI has failed <!-- omit in TOC -->

Reviewer: ```suggestion
### How to Manually Publish the Release to Maven repository  <!-- omit in TOC -->
```

> ⚠️ These following steps are only required if the CI has failed to release the current version in Maven repository.

Reviewer: ```suggestion
⚠️ These following steps should only be applied if it's impossible to release the current version in the Maven repository via the CI. Keep in mind publishing via the CI should always be privileged.
```
